src/main/java/io/kestra/plugin/aws/runner/AwsBatchScriptRunner.java

```diff
@@ -0,0 +1,546 @@
+            try (S3TransferManager transferManager = transferManager(runContext)) {
+                filesToUpload.stream().parallel().map(relativePath ->
+                        UploadFileRequest.builder()
+                            .putObjectRequest(
+                                PutObjectRequest
+                                    .builder()
+                                    .bucket(renderedBucket)
+                                    .key(workingDirKey + "/" + relativePath)
+                                    .build()
+                            )
+                            .source(runContext.resolve(Path.of(relativePath)))
+                            .build()
+                    ).map(transferManager::uploadFile)
+                    .map(FileUpload::completionFuture)
+                    .forEach(throwConsumer(CompletableFuture::get));
+            }
+        }
+
+        containerPropsBuilder.command(command)
+            .jobRoleArn(runContext.render(this.jobRoleArn));
+
+        ComputeEnvironmentDetail computeEnvironmentDetail = client.describeComputeEnvironments(
+                DescribeComputeEnvironmentsRequest.builder()
+                    .computeEnvironments(computeEnvironmentArn)
+                    .build()
+            ).computeEnvironments().stream()
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException("Compute environment not found: " + computeEnvironmentArn));
+
+        PlatformCapability platformCapability = switch (computeEnvironmentDetail.computeResources().type()) {
+            case FARGATE:
+            case FARGATE_SPOT:
+                yield PlatformCapability.FARGATE;
+            case EC2:
+            case SPOT:
+                yield PlatformCapability.EC2;
+            default:
+                yield null;
+        };
+
+        if (platformCapability == PlatformCapability.FARGATE) {
+            containerPropsBuilder.networkConfiguration(
+                NetworkConfiguration.builder()
+                    .assignPublicIp(AssignPublicIp.ENABLED)
+                    .build()
+            );
+        }
+
+
+        logger.debug("Registering job definition");
+        RegisterJobDefinitionResponse registerJobDefinitionResponse = client.registerJobDefinition(
+            RegisterJobDefinitionRequest.builder()
+                .jobDefinitionName(IdUtils.create())
+                .type(JobDefinitionType.CONTAINER)
+                .platformCapabilities(platformCapability)
+                .containerProperties(containerPropsBuilder.build())
+                .build()
+        );
+        String jobDefArn = registerJobDefinitionResponse.jobDefinitionArn();
+        logger.debug("Job definition successfully registered: {}", jobDefArn);
+
+        String jobQueue = runContext.render(this.jobQueueArn);
+        if (jobQueue == null) {
+            logger.debug("Job queue not specified, creating a one-use job queue");
```

Comment:
Is it a one-use job that wil be automatically deleted by AWS or does it miss the queue deletion?

Reply:
The queue deletion is done [here](https://github.com/kestra-io/plugin-aws/blob/feat/aws-batch-script-runner/src/main/java/io/kestra/plugin/aws/runner/AwsBatchScriptRunner.java#L598-L612)